src/Chainweb/Pact/Service/PactQueue.hs

```diff
@@ -51,11 +99,102 @@ addRequest q msg =  atomically $ writeTBQueue priority msg
 
 -- | Get the next available request from the Pact execution queue
 getNextRequest :: PactQueue -> IO RequestMsg
-getNextRequest q = atomically $
+getNextRequest q = do
+  (T2 req entranceTime) <- atomically $
     tryReadTBQueueOrRetry (_pactQueueValidateBlock q)
     <|> tryReadTBQueueOrRetry (_pactQueueNewBlock q)
     <|> tryReadTBQueueOrRetry (_pactQueueOtherMsg q)
+  exitTime <- getCurrentTimeIntegral
+  let requestTime = exitTime `diff` entranceTime
+      stats = case req of
+        ValidateBlockMsg {} -> _pactQueuePactQueueValidateBlockMsgStats q
+        NewBlockMsg {} -> _pactQueuePactQueueNewBlockMsgStats q
+        _ -> _pactQueuePactQueueOtherMsgStats q
+  updatePactQueueStats stats requestTime
+  return req
   where
     tryReadTBQueueOrRetry = tryReadTBQueue >=> \case
       Nothing -> retry
       Just msg -> return msg
+
+
+data PactQueueStats = PactQueueStats
+  {
+    _pactQueueStatsQueueName :: !T.Text
+  , _pactQueueStatsCounters  :: !(IORef PactQueueCounters)
+  }
+
+data PactQueueCounters = PactQueueCounters
+  {
+    _pactQueueCountersCount :: {-# UNPACK #-} !Int
+  , _pactQueueCountersSum   :: {-# UNPACK #-} !Micros
+  , _pactQueueCountersMin   :: {-# UNPACK #-} !Micros
+  , _pactQueueCountersMax   :: {-# UNPACK #-} !Micros
+  } deriving (Show, Generic)
+    deriving anyclass NFData
+
+
+instance ToJSON PactQueueCounters where
+  toJSON (PactQueueCounters {..}) =
```

Comment:
Could you also add an `toEncoding` implementation? You may use the "properties-pattern" that is used for other `ToJSON` instances, too. (This is a new thing and wasn't around when this PR was created)